plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchColumnHandleProperty.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+public final class ElasticsearchColumnHandleProperty
+{
+    private final String name;
```

Comment:
Part of the problem is that we're using the column type (recursively) to drive the instantiation of the decoders. Having this parallel structure introduces potential for the implementation to get out of sync (e.g., what does it mean for the type of a field to be INTEGER and `asRawJson` property to be set to true? Which one wins?

It might be cleaner if we change the abstraction to:
* use the type solely for satisfying the metadata calls and instantiation of column blocks
* keep a nested structure that describes the concrete types of decoders to use

One additional thing to think about is how this relates to "document fields". See `ScanQueryPageSource.flattenFields()` and uses of it.

Reply:
> keep a nested structure that describes the concrete types of decoders to use

@martint does it make sense to still use the nested structure that we're already creating (or the modified version that @phd3 is suggesting with the property path as the key and property stored as the value) and just have another property there (like an enum) that is associated with the Decoders and that will replace what is used in `ScanQueryPageSource.createDecoders()` am I groking that right?

So I would likely generate this when `ElasticsearchMetadat.makeColumnHandles()` gets called if that's the case

Reply:
> Part of the problem is that we're using the column type (recursively) to drive the instantiation of the decoders. Having this parallel structure introduces potential for the implementation to get out of sync 

I think this is addressed now. Decoder type is identified once metadata is fetched and then used while instantiating decoders.

> what does it mean for the type of a field to be INTEGER and asRawJson property to be set to true? Which one wins?

@martint Trino Type in such case would always be `VARCHAR` (returned from `toTrinoType`), as that method is also aware of Metadata stored in the `IndexMetadata.Field`.  If there're issues with reading it as `VARCHAR`, the scan would fail. Does this seem reasonable or did you have something else in mind?  

> One additional thing to think about is how this relates to "document fields". See ScanQueryPageSource.flattenFields() and uses of it.

I'm not sure how this method would help here. @martint can you please elaborate? Based on the above logic, the visible type would be VARCHAR, if an original TIMESTAMP_MILLIS field or any of its parents were marked asRawJson.